File: SRC/xerbla.f
      void xerbla(SRNAME, INFO ) {

// -- LAPACK auxiliary routine --
// -- LAPACK is a software package provided by Univ. of Tennessee,    --
// -- Univ. of California Berkeley, Univ. of Colorado Denver and NAG Ltd..--

      // .. Scalar Arguments ..
      List<String>         SRNAME;
      int                INFO;
      // ..

// =====================================================================

      // .. Intrinsic Functions ..
      // INTRINSIC LEN_TRIM
      // ..
      // .. Executable Statements ..

      WRITE( *, FMT = 9999 )SRNAME( 1:LEN_TRIM( SRNAME ) ), INFO;

      STOP;

 9999 FORMAT( ' ** On entry to ', A, ' parameter number ', I2, ' had ', 'an illegal value' );

      // End of XERBLA

      }

File: lib/src/xerbla.dart
// Copyright (c) 2024 Guilherme Lepsch. All rights reserved. Use of this
// source code is governed by a BSD-style license that can be found in the
// [LICENSE file](https://github.com/lepsch/dart_lapack/blob/main/LICENSE).

void Function(String SRNAME, int INFO) xerbla = _xerbla;

void _xerbla(final String SRNAME, final int INFO) {
  throw ArgumentError(
      ' ** On entry to $SRNAME parameter number $INFO had an illegal value');
}
